ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/party_management_service.proto

```diff
@@ -111,28 +125,63 @@ message AllocatePartyRequest {
 
   // Human-readable name of the party to be added to the participant. It doesn't
   // have to be unique.
+  // Use of this field is discouraged. Use ``local_metadata`` instead.
   // Optional
   string display_name = 2;
+
+  // Participant-local metadata to be stored in the ``PartyDetails`` of this newly allocated party.
+  // Optional
+  ObjectMeta local_metadata = 3;
 }
 
 message AllocatePartyResponse {
 
   PartyDetails party_details = 1;
 }
 
+// Required authorization: ``HasRight(ParticipantAdmin)``
+message UpdatePartyDetailsRequest {
+  // Party to be updated
+  //
+  // Required
+  // Modifiable
+  PartyDetails party_details = 1;
+
+  // Specifies the fields of the party details to be updated.
+  // At least one field must be specified, e.g. 'party_details.local_metadata'.
+  //
+  // Required
+  google.protobuf.FieldMask update_mask = 2;
+}
+
+message UpdatePartyDetailsResponse {
+  // Updated party details
+  PartyDetails party_details = 1;
+}
+
 message PartyDetails {
 
   // The stable unique identifier of a Daml party.
   // Must be a valid PartyIdString (as described in ``value.proto``).
+  //
   // Required
   string party = 1;
 
-  // Human readable name associated with the party. Caution, it might not be
-  // unique.
+  // Human readable name associated with the party at allocation time.
+  // Caution, it might not be unique.
+  // Use of this field is discouraged. Use the `local_metadata` field instead.
+  //
   // Optional
   string display_name = 2;
 
   // true if party is hosted by the backing participant.
-  // Required
+  //
+  // Optional
   bool is_local = 3;
+
+  // Participant-local metadata of this party.
```

Comment:
```suggestion
  // Participant-local metadata of this party. Only set if is_local = true.
```

Reply:
Why should we limit this to 'local' parties?

Reply:
It's participant local metadata. How can you expose this for non-local parties?

Reply:
'non-local' means they were allocated by a different participant, but the participant knows about them the same as it knows by parties allocated by itself: for both there are entries in party_details table in the indexdb.

Reply:
Oh so I can add metadata for a party from another participant? That seems counter intuitive to me.

Reply:
A use case could be setting a display name like annotation, e.g. on my participant node I could name Alice's 'painter' party 'Alice's painter' to discriminate from my own 'painter' party.

Reply:
Do we have a concrete requirement or usecase for this? For the display name, I don’t entirely buy it. I think we’re better off solving display names at an application level rather than trying to bake this into the participant.

Reply:
The general principle here would be "metadata for participant-managed entities". Parties from other participants are not managed via your participant so you also don’t get metadata for those.

Reply:
@da-tanabe @danielporterda @davidrichards-da What's your take on allowing metadata only for 'local' parties, i.e. allocated by the current participant node?

Reply:
@da-tanabe I've just noticed this unanswered. Are you able to add an answer here? Is this still relevant @pbatko-da?

Reply:
It's still a relevant design decision it'd be good to get Hub's stance on.

For the current implementation (PR will be ready for review soon) I assumed that both local and non-local parties can have metadata assigned.